apps/climatemappedafrica/src/payload/fields/LocationSelect.js

```diff
@@ -0,0 +1,43 @@
+import { Select } from "payload/components/forms";
+import { select } from "payload/dist/fields/validations";
+import { createElement, useMemo } from "react";
+import useSWR from "swr";
+
+const fetcher = (url) => fetch(url).then((res) => res.json());
+const apiUrl = process.env.PAYLOAD_PUBLIC_APP_URL;
+
+export async function validateLocation(value, { hasMany, required, t }) {
+  const response = await fetch(`${apiUrl}/api/hurumap/profiles`);
+  const data = await response.json();
+  const { locations } = data ?? {};
+  const options =
+    locations
+      ?.filter(({ level }) => level !== "region")
+      ?.map((location) => ({
+        label: location.name,
+        value: location.code,
+      })) || [];
+  return select(value, { hasMany, options, required, t });
+}
+
+function LocationSelect(props) {
+  const url = `${apiUrl}/api/hurumap/profiles`;
+  const { data } = useSWR(url, fetcher, {
+    dedupingInterval: 60000,
+    revalidateOnFocus: false,
+  });
+  const { locations } = data ?? {};
+  const memoOptions = () =>
+    (
+      locations
+        ?.filter(({ level }) => level !== "region")
```

Comment:
Does this work across counties? i.e. won't we end up with counties or provinces as valid options?

Reply:
Yes @kilemensi all locations under country have `level: region` key value pair  whether province county etc.